### PR TITLE
Fix issues of calling WAMR API's

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

File: pservice/CMakeLists.txt [hunk 1]
```diff
@@ -65,7 +65,8 @@ SET(SGX_INCLUDE "${SGX_SDK}/include")
 SET(SGX_SSL_INCLUDE "${SGX_SSL}/include")
 SET(SGX_ENCLAVE_INCLUDE "${SGX_SDK}/include"
                         "${SGX_SDK}/include/tlibc"
-                        "${SGX_SDK}/include/libcxx")
+                        "${SGX_SDK}/include/libcxx"
+                        "${SGX_SSL}/include")
```
Comment (cmickeyb) [inline pservice/CMakeLists.txt:70]: why are these necessary? we shouldn't be including anything directly from SSL. It should only be included through common.

File: common/interpreter/wawaka_wasm/WawakaInterpreter.cpp [hunk 2]
```diff
@@ -192,9 +198,19 @@ int32 WawakaInterpreter::evaluate_function(
...
+    if (argv[0] == 0) {
+        SAFE_LOG(PDO_LOG_ERROR, "module malloc failed for some reason");
+        return 0;
+    }
+
+    argv[1] = buf_offset1 = (int32)wasm_runtime_module_malloc(wasm_module_inst, env.length() + 1);
+    if (argv[1] == 0) {
+        SAFE_LOG(PDO_LOG_ERROR, "module malloc failed for some reason");
+        return 0;
+    }
```
Comment (cmickeyb) [inline common/interpreter/wawaka_wasm/WawakaInterpreter.cpp:214]: we definitely need to free the allocated buffers. would prefer to put this in a try/catch/finally block. for example, this code doesn't free buffer 0 if buffer 1 fails. and we generally throw exceptions on allocation errors. 

final question... do we need to free the result once we copy it out? presumably... since we destroy the interpreter at the end of the execution, everything is free'd anyway.

**Review by cmickeyb — CHANGES_REQUESTED** [review]
Mostly straightforward.

File: pservice/CMakeLists.txt (hunk 1 above)
Comment (masomel) [reply pservice/CMakeLists.txt:70]: I think we can skip these changes. PDO builds fine without them.

Comment (wenyongh) [reply pservice/CMakeLists.txt:70]: OK. But it is strange that it reports error when building eservice if I don't add the SSG_SSL/include to include path:
[ 35%] Building CXX object lib/libpdo_enclave/CMakeFiles/libpdo-enclave.dir/contract_enclave.cpp.o
In file included from /home/agent/WASM/private-data-objects/eservice/lib/libpdo_enclave/contract_code.cpp:22:0:
/home/agent/WASM/private-data-objects/eservice/../common/crypto/crypto.h:18:10: fatal error: openssl/sha.h: No such file or directory
 #include <openssl/sha.h>

File: common/interpreter/wawaka_wasm/WawakaInterpreter.cpp (hunk 2 above)
Comment (wenyongh) [reply common/interpreter/wawaka_wasm/WawakaInterpreter.cpp:214]: Sorry for not freeing buffer 0 if buffer 1 fails, you should add it.

The buffer is allocated from the heap space, which is initialized int module instantiating and will be destroyed when module instance is de-instantiated. So no memory leak will happen if you don't free them, the issue is that, if you keep calling module_malloc, it may fail when the heap is used up. So it you copy the buffer out, you don't need to free it.

And if module_malloc fails, the runtime throws exception: out of memory.

File: pservice/CMakeLists.txt (hunk 1 above)
Comment (wenyongh) [reply pservice/CMakeLists.txt:70]: Is is up to you. I can manually modify the makefile to make it build success.

**Comment by masomel** [comment]
@wenyongh This PR is now being pulled in as part of https://github.com/hyperledger-labs/private-data-objects/pull/249 given the latest updates to WAMR.
